src/Microsoft.Azure.ServiceBus/Amqp/AmqpSubscriptionClient.cs

```diff
@@ -142,5 +145,46 @@ public Task CloseAsync()
                 throw AmqpExceptionHelper.GetClientException(exception);
             }
         }
+
+        public async Task<IEnumerable<RuleDescription>> OnGetRulesAsync(int top, int skip)
+        {
+            try
+            {
+                var amqpRequestMessage =
+                    AmqpRequestMessage.CreateRequest(
+                        ManagementConstants.Operations.EnumerateRulesOperation,
+                        this.ServiceBusConnection.OperationTimeout,
+                        null);
+                amqpRequestMessage.Map[ManagementConstants.Properties.Top] = top;
+                amqpRequestMessage.Map[ManagementConstants.Properties.Skip] = skip;
+
+                var response = await this.InnerReceiver.ExecuteRequestResponseAsync(amqpRequestMessage).ConfigureAwait(false);
+                var rules = new List<RuleDescription>();
+                if (response.StatusCode == AmqpResponseStatusCode.OK)
+                {
+                    var ruleList = response.GetListValue<AmqpMap>(ManagementConstants.Properties.Rules);
+                    foreach (var entry in ruleList)
+                    {
+                        var amqpRule = (AmqpMap)entry[ManagementConstants.Properties.RuleDescription];
+                        var ruleDescription = AmqpMessageConverter.GetRuleDescription(amqpRule);
+                        rules.Add(ruleDescription);
+                    }
+                }
+                else if (response.StatusCode == AmqpResponseStatusCode.NoContent)
+                {
+                    rules = new List<RuleDescription>();
```

Comment:
+1, not needed right?

---
In reply to: [123424848](https://github.com/Azure/azure-service-bus-dotnet/pull/192#discussion_r123424848) [](ancestors = 123424848)